targets/TARGET_Freescale/TARGET_KLXX/us_ticker.c

```diff
@@ -187,11 +187,11 @@ static void lptmr_isr(void) {
 void us_ticker_set_interrupt(timestamp_t timestamp) {
     int delta = (int)((uint32_t)timestamp - us_ticker_read());
     if (delta <= 0) {
-        // This event was in the past:
-        us_ticker_irq_handler();
-        return;
-    }
-    
+        // This event was in the past.  Force it into the very near
```

Comment:
It is undecidable to know if an event if in the past or in the future because the counter **wrap**. 
Given that information `int delta = (int)((uint32_t)timestamp - us_ticker_read());` can be negative for a lot of valid value (whenever `timestamp >= us_ticker_read() + 0x80000000` ).